front/components/workspace/BuyCreditDialog.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  Dialog,
+  DialogContainer,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  Input,
+} from "@dust-tt/sparkle";
+import { useEffect, useState } from "react";
+
+import { useSendNotification } from "@app/hooks/useNotification";
+import { purchaseCredits } from "@app/lib/client/credits";
+import { useSubmitFunction } from "@app/lib/client/utils";
+
+interface BuyCreditDialogProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSuccess?: () => void;
+  workspaceId: string;
+  isEnterprise: boolean;
+}
+
+export function BuyCreditDialog({
+  isOpen,
+  onClose,
+  onSuccess,
+  workspaceId,
+  isEnterprise,
+}: BuyCreditDialogProps) {
+  const [amountDollars, setAmountDollars] = useState<string>("");
+  const sendNotification = useSendNotification();
+
+  // Reset state when dialog opens
+  useEffect(() => {
+    if (isOpen) {
+      setAmountDollars("");
+    }
+  }, [isOpen]);
+
+  const { submit: handlePurchase, isSubmitting } = useSubmitFunction(
+    async () => {
+      const amount = parseFloat(amountDollars);
+      const success = await purchaseCredits({
+        workspaceId,
+        amountDollars: amount,
+        sendNotification,
+      });
+
+      if (success) {
+        onClose();
+        onSuccess?.();
+      }
+    }
+  );
+
+  return (
+    <Dialog open={isOpen} onOpenChange={(open) => !open && onClose()}>
+      <DialogContent size="md">
+        <DialogHeader>
+          <DialogTitle>Purchase Programmatic Credits</DialogTitle>
+          <DialogDescription>
+            Purchase credits for programmatic API usage and integrations.
+          </DialogDescription>
+        </DialogHeader>
+        <DialogContainer>
+          <div className="flex flex-col gap-4">
+            <div className="flex flex-col gap-2">
+              <label
+                htmlFor="amount"
+                className="text-element-800 text-sm font-medium"
+              >
+                Amount (USD)
+              </label>
+              <Input
+                id="amount"
+                type="number"
+                placeholder="10.00"
+                value={amountDollars}
+                onChange={(e) => setAmountDollars(e.target.value)}
+                min="0"
+                step="0.01"
```

Comment:
Just curious why step "0.01"? (feeling a bit weird that you can purchase such a granular number of credits but I don't have context)